ginga/rv/plugins/LineProfile.py

```diff
@@ -248,25 +258,43 @@ def _plot(self, mark=None):
         mddata = self.image.get_mddata().T
         naxes = mddata.ndim
 
-        if self.selected_axis:
-            plot_x_axis_data = self.get_axis(self.selected_axis)
-            if plot_x_axis_data is None:
-                # image may lack the required keywords, or some trouble
-                # building the axis
-                return
-            slice_obj = self._slice(naxes, mk=mark)
-            plot_y_axis_data = mddata[slice_obj]
+        if not self.selected_axis:
+            return
 
-            self.clear_plot()
-            self.plot.plot(plot_x_axis_data, plot_y_axis_data,
-                           xtitle=self.x_lbl, ytitle=self.y_lbl)
+        plot_x_axis_data = self.get_axis(self.selected_axis)
+        if plot_x_axis_data is None:
+            # image may lack the required keywords, or some trouble
+            # building the axis
+            return
+        slice_obj = self._slice(naxes, mk=mark)
+        plot_y_axis_data = mddata[slice_obj]
 
+        # If few enough data points, add marker
+        if len(plot_y_axis_data) <= 10:
+            marker = 'o'
         else:
-            self.fv.show_error("Please select an axis")
+            marker = None
+
+        # https://github.com/matplotlib/matplotlib/issues/3633/
+        ax2 = self._ax2
+        ax2.patch.set_visible(False)
+
+        self.clear_plot()
+        self.plot.plot(plot_x_axis_data, plot_y_axis_data,
+                       xtitle=self.x_lbl, ytitle=self.y_lbl, marker=marker)
+
+        # Top axis to show pixel location across X
+        ax2.cla()
+        xx1, xx2 = self.plot.ax.get_xlim()
+        ax2.set_xlim((xx1 - self._crval) / self._cdelt,
```

Comment:
This is so that I know where exactly in the cube that a feature of interest appears. Just showing the velocity is not very useful to pinpoint location in cube at a glance without playing through `MultiDim`.